### DISCUSSION TITLE
FirebasePerformance as dynamic framework

### DISCUSSION THREAD

**Discussion by ariel-demarco-peya** [firebase/firebase-ios-sdk#7569] [Pitches]
Hi!
I'm working on some private pods (cocoapods) and read that many Firebase pods are dynamic now. Those are great news as we prefer to have dynamic frameworks. I updated my Firebase dependencies to version 7.x in my pod, and when I tried to publish a new version of the pod I received this error:
`- ERROR | [iOS] unknown: Encountered an unknown error (The 'Pods-App' target has transitive dependencies that include statically linked binaries: (FirebasePerformance)) during validation.`

It seems that FirebasePerformance [is static by default](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebasePerformance.podspec#L27). Is there any reason why that framework wasn't included in the [original change](https://github.com/firebase/firebase-ios-sdk/pull/6557)? I know the quick fix is to keep using the `static_framework = true` in my pod too, but I wanted to know if there's a roadmap to change that or there's a specific reason why it's static.

**Comment by paulb777** [comment]
HI @ariel-demarco-peya Thanks for the report. This was an oversight from work happening on different git branches at the time. I'll send a PR to fix for Firebase 7.8.0.

**Reply by paulb777** [reply]
PR at #7570

**Reply by ariel-demarco-peya** [reply]
Great! Thanks for the quick response (and PR 😂 )
